tests: Fix ESP8266 test failures.

agatti opened the pull request:
### Summary

This PR contains some small changes to the test suite to let test runs complete cleanly on ESP8266.

* `extmod/machine_uart_tx.py` would fail with an `ImportError` since there was no ESP8266 wiring module.  An appropriate module is now present and the test was updated with ESP8266-specific timing data
* `extmod/vfs_blockdev_invalid.py` would run out of memory when executing the second half of the test.  The test was rewritten to delay any success output until validation is performed, returning the appropriate marker in case of a low memory condition
* `stress/list_sort.py` would take too long to complete, triggering the OS watchdog.  The test is now skipped as part of the ESP8266 test run.

### Testing

Besides the regular CI tests, the test suite executed without special options now completes with no errors on an ESP8266EX board.

agatti committed:
tests/run-tests.py: Skip list sort stress test for ESP8266.

This commit marks the "stress/list_sort.py" test to be skipped when
running on ESP8266.

The test just takes too long without yielding to the OS whilst doing the
sort, causing the internal software watchdog to kick in and reboot the
board.

Signed-off-by: Alessandro Gatti <a.gatti@frob.it>

agatti committed:
tests/extmod/vfs_blockdev_invalid.py: Handle low memory conditions.

This commit modifies the "extmod/vfs_blockdev_invalid" test to better
behave on boards with low available memory.

Before these changes the test would fail on ESP8266 (at least), due to
low memory, but in a way that could not be easily solved as the error
occurred in the middle of the test.

The test has been rewritten to delay its output until the very end, so
if a low memory condition occurs and needs to stop execution then no
real output will show up before the skip marker.

Signed-off-by: Alessandro Gatti <a.gatti@frob.it>

dpgeorge commented on tests/extmod/machine_uart_tx.py:
```diff
@@ -3,12 +3,12 @@
 
 try:
     from machine import UART
+    from target_wiring import uart_loopback_args, uart_loopback_kwargs
```
It's possible to use the existing `bit_margin` and `timing_margin_us` parameters in the test to tweak the test for esp8266.  Just add a new case for `sys.platform == "esp8266"`.  At least then the test does something useful, even if it's quite lenient for this platform.

agatti committed:
tests/target_wiring: Provide an ESP8266 target wiring module.

This commit introduces ESP8266 support for target wiring tests, fixing
execution of relevant tests that once failed on that platform.

ESP8266 boards need to have GPIO4 and GPIO5 connected together to
provide a UART loopback, in order to test whether UART data effectively
flows through.

The wiring-enabled UART transmission timing test was also updated with
measurements compatible with a few ESP8266 test boards.

Signed-off-by: Alessandro Gatti <a.gatti@frob.it>

dpgeorge reviewed (approved):
Thanks for this.  I see these failures as part of the testing I do for releases and it was on my list to fix them.  But now it's done!
